frontend/src/metabase/visualizations/echarts/pie/format.ts

```diff
@@ -0,0 +1,48 @@
+import type {
+  ComputedVisualizationSettings,
+  RenderingContext,
+} from "metabase/visualizations/types";
+import { computeMaxDecimalsForValues } from "metabase/visualizations/lib/utils";
+
+import type { PieChartModel } from "./model/types";
+
+export interface PieChartFormatters {
+  formatMetric: (value: unknown) => string;
+  formatPercent: (value: unknown) => string;
+}
+
+export function getPieChartFormatters(
+  chartModel: PieChartModel,
+  settings: ComputedVisualizationSettings,
+  renderingContext: RenderingContext,
+): PieChartFormatters {
+  const { column: getColumnSettings } = settings;
+  if (!getColumnSettings) {
+    throw Error(`"settings.column" is undefined`);
+  }
+  const metricColSettings = getColumnSettings(
+    chartModel.colDescs.metricDesc.column,
+  );
+
+  const formatMetric = (value: unknown) =>
+    renderingContext.formatValue(value, {
+      ...metricColSettings,
+    });
+
+  const formatPercent = (value: unknown) =>
+    renderingContext.formatValue(value, {
+      column: metricColSettings.column,
+      number_separators: metricColSettings.number_separators,
+      number_style: "percent",
+      jsx: true,
```

Comment:
I left a comment about it in a previous PR but feel free to just drop `jsx` here since the formatters got moved